base/iterators.jl

```diff
@@ -1343,4 +1344,144 @@ only(x::NamedTuple) = throw(
     ArgumentError("NamedTuple contains $(length(x)) elements, must contain exactly 1 element")
 )
 
+
+"""
+    logrange(start => stop, length)
+
+Returns an iterator which runs from `start` to `stop` with `length` elements
+spaced logarithmically, rather than linearly as for [`range`](@ref).
+That is, the ratio of successive elements is constant, not the difference.
+
+!!! compat "Julia 1.7"
+    This function requires at least Julia 1.7.
+
+This is similar to `geomspace` in Python, and to `PowerRange` in Mathematica.
+
+# Examples
+```jldoctest
+julia> foreach(println, logrange(2 => 16, 4))
+2.0
+4.0
+8.0
+16.0
+
+julia> collect(logrange(1000 => 1, 7))
+7-element Vector{Float64}:
+ 1000.0
+  316.22776601683796
+  100.0
+   31.622776601683793
+   10.0
+    3.1622776601683795
+    1.0
+
+julia> ans ≈ 10 .^ (3:-0.5:0)
+true
+
+julia> collect(logrange(-1 => -2f0, 3))
+3-element Vector{Float32}:
+ -1.0
+ -1.4142135
+ -2.0
+
+julia> map(rad2deg∘angle, logrange(1 => -1+0im, 5))
+5-element Vector{Float64}:
+   0.0
+  45.0
+  90.0
+ 135.0
+ 180.0
+```
+"""
+function logrange(fromto::Pair{<:Number, <:Number}, len::Integer)
+    len >= 2 || throw(ArgumentError("logrange must have length of at least 2"))
+    start, stop = map(float, promote(fromto...))
+    ratio = ratio_nth_root(stop, start, Int(len)-1)
+    LogRange(start, ratio, Int(len))
+end
+
+ratio_nth_root(a::Number, b::Number, n::Int) = (a/b)^(1/n)
+ratio_nth_root(a::Float32, b::Float32, n::Int) = (Float64(a)/Float64(b))^(1/n)
+ratio_nth_root(a::ComplexF32, b::ComplexF32, n::Int) = (ComplexF64(a)/ComplexF64(b))^(1/n)
+function ratio_nth_root(a::T, b::T, m::Int) where {T<:Union{Float64, ComplexF64}}
+    over = Base.TwicePrecision(a) / Base.TwicePrecision(b)
+    r1 = Base.TwicePrecision((over.hi)^(1/m))
+    # Refine that using one step of Newton's method:
+    r1pow = prod(r1 for _ in 1:m-1)
+    r2 = (m-1)*r1/m + over / (m * r1pow)
+    # r2pow = prod(r2 for _ in 1:m-1)
+    # r3 = (m-1)*r2/m + over / (m * r2pow)
+end
```

Comment:
I recommend calling this form of `ratio_nth_root` something else with more permissive typing. You can then invoke from `ComplexF64` code which needs to be moved to complex.jl as well as `Float64` code in range.jl.

Reply:
Types aren't finalised here; the higher-precision stuff isn't working so well for complex numbers at the moment, I meant to look more closely.

Reply:
OK, I investigated. Iterating Newton's method with `TwicePrecision{Float64}`, the relative error bottoms out at about `1e-22`, which is invisible in `Float64`. 

But with `TwicePrecision{ComplexF64}`, instead it stops at about `1e-11`, and hence the real part of e.g. `logrange(1 => -1 + 0im, 8+1) |> collect` still shows an error. Using complex `BigFloat` instead gets us to `1e-78`, which can make the real part perfect -- but still not the imaginary part, since `-1.0 + 1e-79im` is a perfectly cromulent `ComplexF64`. 

So I give up on making complex numbers exact.